### PR TITLE
Add missing cloud-init config flag for reset

### PR DESCRIPTION
**Pull request by davidcassany**

### CODE REVIEW

**Comment by davidcassany** [comment]
draft to check if unit tests pass and because I am still thinking on making a shared structure with common fields of reset and install specs, there is quite some overlap already.

File: pkg/constants/constants.go [hunk 1]
```diff
@@ -195,10 +195,12 @@ func GetInstallKeyEnvMap() map[string]string {
...
-		"target":          "TARGET",
-		"system.uri":      "SYSTEM",
-		"tty":             "TTY",
-		"grub-entry-name": "GRUB_ENTRY_NAME",
+		"system.uri":       "SYSTEM",
+		"tty":              "TTY",
+		"grub-entry-name":  "GRUB_ENTRY_NAME",
+		"cloud-init":       "CLOUD_INIT",
+		"reset-persistent": "PERSISTENT",
+		"reset-oem":        "OEM",
```
Comment (davidcassany) [inline pkg/constants/constants.go:203]: These will translate in to "ELEMENTAL_RESET_*" variables. For instance `reset-oem` flag will be mapped as `ELEMENTAL_RESET_OEM` variable and `cloud-init` to `ELEMENTAL_RESET_CLOUD_INIT`, etc.

**Review by anmazzotti — APPROVED** [review]
Looks good, thank you for the change!

**Comment by anmazzotti** [comment]
@davidcassany may you also touch the documentation? https://github.com/rancher/elemental-toolkit/blob/main/docs/elemental_reset.md

I'm going to reference it in the docs

**Comment by davidcassany** [comment]
> @davidcassany may you also touch the documentation? https://github.com/rancher/elemental-toolkit/blob/main/docs/elemental_reset.md

Those used to be autogenerated, however I am not fully convinced the bot survived the latest refactor around elemental-toolkit and elemental-cli

**Comment by anmazzotti** [comment]
Resolves rancher/elemental#526

**Pull request merged**